### PR TITLE
Query categoryTerms from Graphql

### PR DESCRIPTION
**Pull request by josephgregoryii**
### Summary
Earlier this sprint, we had a [bug](https://issues.newrelic.com/browse/NR-3896) in the internal I/O where dashboards and alerts were not rendering under associated tabs. Specifically, the `partner` category.

We had originally used a [hard coded file](https://github.com/newrelic/newrelic-quickstarts/blob/0421c87883c3f35c6ac3449971fc412de1d05e69/utils/instant-observability-categories.ts) to match the category terms with the associated tabs for quickstarts. Before a fix, the categories file was outdated and needed an update, along with mutations submitted to graphql to update those quickstarts.

This PR allows us to fetch the category terms during submission to avoid losing category terms in the future, while also removing the category files

### CODE REVIEW

File: utils/lib/Quickstart.ts [hunk 1]
```diff
@@ -125,9 +125,11 @@ class Quickstart {
 
   /**
    * Get mutation variables from quickstart config
-   * @returns - mutation varaibles for quickstart
+   * @returns - Promised mutation variables for quickstart
    */
-  getMutationVariables(dryRun: boolean): QuickstartMutationVariable {
+  async getMutationVariables(
```
Comment (aswanson-nr) [inline utils/lib/Quickstart.ts:130]: We'll want to make sure that is function is now being awaited wherever it's used throughout the scripts

File: utils/lib/nr-graphql-helpers.ts [hunk 2]
```diff
@@ -124,10 +124,22 @@ export const translateMutationErrors = (
   }
 };
 
+type CategoryTermsNRGraphqlResults = {
+  actor: {
+    nr1Catalog: {
+      categories: {
+        displayName: string;
+        slug: string;
```
Comment (aswanson-nr) [inline utils/lib/nr-graphql-helpers.ts:132]: Are these fields even needed? On lines 165-167 we are just flat mapping down to the terms.

**Review by aswanson-nr — COMMENTED** [review]
I found a few things worth checking on, but overall looks fantastic!

File: utils/lib/Quickstart.ts (hunk 1 above)
Comment (josephgregoryii) [reply utils/lib/Quickstart.ts:130]: Weird, i must've restored at some point and lost the changes i made. I'll go back and fix this!

File: utils/lib/nr-graphql-helpers.ts (hunk 2 above)
Comment (josephgregoryii) [reply utils/lib/nr-graphql-helpers.ts:132]: We do not - should i update the query to reflect this as well?

File: utils/lib/Quickstart.ts (hunk 1 above)
Comment (josephgregoryii) [reply utils/lib/Quickstart.ts:130]: Only needed the fix inside `create-quickstarts-snapshot.ts` 👍🏻 . `create_validate_pr_quickstart` has the function calls mapped inside a `Promise.all`

File: utils/lib/nr-graphql-helpers.ts (hunk 2 above)
Comment (josephgregoryii) [reply utils/lib/nr-graphql-helpers.ts]: updated on the typescript side of things 👍🏻

Comment (aswanson-nr) [reply utils/lib/nr-graphql-helpers.ts]: Might as well remove those from the query as well, no need to request more than we need!

**Review by aswanson-nr — APPROVED** [review]

**Review by zstix — APPROVED** [review]

**Pull request merged**